packages/api/endpoints/collections.js

```diff
@@ -197,32 +198,42 @@ async function put(req, res) {
   }
 
   try {
-    oldCollection = await collectionsModel.get({ name, version });
+    oldPgCollection = await collectionPgModel.get(knex, { name, version });
   } catch (error) {
     if (error.name !== 'RecordDoesNotExist') {
       throw error;
     }
     return res.boom.notFound(`Collection '${name}' version '${version}' not found`);
   }
 
+  try {
+    oldCollection = await collectionsModel.get({ name, version });
+  } catch (error) {
+    if (error.name !== 'RecordDoesNotExist') {
+      throw error;
+    }
+    log.warn(`Dynamo record for Collection '${name}' version '${version}' not found, proceeding to update with postgresql record alone`);
+  }
+
   collection.updatedAt = Date.now();
-  collection.createdAt = oldCollection.createdAt;
+  collection.createdAt = oldPgCollection.created_at.getTime();
 
   const postgresCollection = translateApiCollectionToPostgresCollection(collection);
 
   try {
     await createRejectableTransaction(knex, async (trx) => {
-      const [pgCollection] = await collectionPgModel.upsert(trx, postgresCollection);
-      const translatedCollection = await translatePostgresCollectionToApiCollection(pgCollection);
+      await collectionPgModel.upsert(trx, postgresCollection);
       dynamoRecord = await collectionsModel.create(collection);
       // process.env.ES_INDEX is only used to isolate the index for
       // each unit test suite
       await indexCollection(esClient, dynamoRecord, process.env.ES_INDEX);
-      await publishCollectionUpdateSnsMessage(translatedCollection);
+      await publishCollectionUpdateSnsMessage(collection);
     });
   } catch (error) {
     // Revert Dynamo record update if any write fails
-    await collectionsModel.create(oldCollection);
+    if (oldCollection) {
```

Comment:
Wondering if we need to publish to SNS in these cases of "reverting"...

Thinking through it, when this happens, we're just reverting the record back to what it was. And presumably an SNS message was sent when the record was put into its previous state, so I don't think we need to an SNS message for these cases